docs/quickstarts/csharp-quickstart.md

```diff
@@ -30,35 +35,75 @@ $ # Clone the repository to get the example code:
 $ git clone -b {{ site.data.config.grpc_release_branch }} https://github.com/grpc/grpc 
 ```
 
-All of the files in this quickstart are in the directory `examples/csharp/helloworld`.
+#### Visual Studio, Xamarin Studio, and Mondevelop IDEs
+
+* The examples are in the directory, `examples/csharp/helloworld`.
+
+#### .NET Core SDK
 
-The example solution in this walkthrough already adds the necessary
+* A .NET Core SDK version of the hello world examples are in the directory, `examples/csharp/helloworld-from-cli`.
+
+The example in this walkthrough already adds the necessary
 dependencies for you (Grpc, Grpc.Tools and Google.Protobuf NuGet packages).
 
 ## Build the example
 
-* Open the solution `Greeter.sln` with Visual Studio, Monodevelop (on Linux) or Xamarin Studio (on Mac OS X)
+### Visual Studio
+* Open the solution `Greeter.sln` with Visual Studio.
 * Build the solution (this will automatically download NuGet dependencies)
+
+### Using .NET Core SDK from the command line
+From the `examples/csharp/helloworld-from-cli` directory:
+
+```
+> dotnet restore
+> dotnet build **/project.json
+```
+
+### Xamarin Studio and Monodevelop
+The C# gRPC package currently depends on System.Interactive.Async 3.0.0, which cannot be downloaded on older NuGet versions.
+NuGet is too old in Xamarin Studio on OS X and Monodevelop on Linux as well as via apt-get.
+
+One possible workaround is as follows:
+
+* Upgrade your NuGet installation with `/path/to/nuget update -self`. 
+* From the `examples/csharp/helloworld` directory, run `/path/to/nuget restore`. 
+* Now that the NuGet dependencies are restored into their proper package folders, build
+  the solution from your IDE.
   
 ## Run a gRPC application
 
+### After building with Visual Studio, Xamarin Studio, or Monodevelop IDEs:
```

Comment:
This and the rest of the headers aren't seeming like headers so much. If you're putting a colon at the end of a header, it probably isn't a header, or it needs to be reworded.

So, for example, here you could have:

```
### Using Visual Studio, Xamarin Studio, or Monodevelop IDEs

After building, from the ...
```